view/src/main/java/dev/marcellogalhardo/retained/view/ViewRetainedObject.kt

```diff
@@ -0,0 +1,52 @@
+package dev.marcellogalhardo.retained.view
+
+import android.app.Activity
+import android.os.Bundle
+import android.view.View
+import androidx.core.os.bundleOf
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ViewModelStoreOwner
+import androidx.lifecycle.findViewTreeLifecycleOwner
+import androidx.lifecycle.findViewTreeViewModelStoreOwner
+import androidx.navigation.NavBackStackEntry
+import androidx.savedstate.SavedStateRegistryOwner
+import androidx.savedstate.findViewTreeSavedStateRegistryOwner
+import dev.marcellogalhardo.retained.core.InternalRetainedApi
+import dev.marcellogalhardo.retained.core.RetainedEntry
+import dev.marcellogalhardo.retained.core.createRetainedObjectLazy
+
+/**
+ * Returns a [Lazy] delegate to access a retained object by **default** scoped to this
+ * [View] (e.g., [NavBackStackEntry], [Fragment] or [Activity]).
+ *
+ * ```
+ * class MyView(context: Context) : View(context) {
+ *     val vm by retain { ViewModel() }
+ * }
+ * class ViewModel(val name: String = "")
+ * ```
+ *
+ * This property can be accessed only after the [LifecycleOwner] is ready to use Jetpack
+ * [ViewModel], and access prior to that will result in [IllegalArgumentException]. TODO: right now it actually throws a NPE
```

Comment:
Any idea why it is throwing a NPE? It looks like it should work.

Reply:
Oh I just copy pasted the docs from the other methods in the library and changed the minimum I needed, but decided to leave this untouched and just add the `TODO` to decide later whether the NPE is good enough or if we want to throw something different.

It's NPE in this case because of the nature of the APIs we're using here: they are all nullable. If you look at the implementation you'll see plenty of beautiful double bangs. They are the cause for the NPE.